src/app/server/java/ChipFabricProvider-JNI.cpp

```diff
@@ -0,0 +1,130 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+/**
+ *    @file
+ *      Implementation of JNI bridge for CHIP App Server for Android TV apps
+ *
+ */
+#include "ChipFabricProvider-JNI.h"
+#include "AndroidAppServerWrapper.h"
+#include <app/server/Server.h>
+#include <cstdlib>
+#include <iostream>
+#include <jni.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/CHIPJNIError.h>
+#include <lib/support/JniTypeWrappers.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <thread>
+#include <trace/trace.h>
+
+using namespace chip;
+
+#define JNI_METHOD(RETURN, METHOD_NAME) extern "C" JNIEXPORT RETURN JNICALL Java_chip_appserver_ChipFabricProvider_##METHOD_NAME
+
+namespace {
+JavaVM * sJVM;
+// jclass sFabricProviderCls = NULL;
+} // namespace
+
+CHIP_ERROR AndroidChipFabricProviderJNI_OnLoad(JavaVM * jvm, void * reserved)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env;
+
+    ChipLogProgress(DeviceLayer, "ChipFabricProvider JNI_OnLoad() called");
+
+    chip::Platform::MemoryInit();
+
+    // Save a reference to the JVM.  Will need this to call back into Java.
+    JniReferences::GetInstance().SetJavaVm(jvm, "chip/appserver/ChipFabricProvider");
+    sJVM = jvm;
+
+    // Get a JNI environment object.
+    env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrExit(env != NULL, err = CHIP_JNI_ERROR_NO_ENV);
+
+    ChipLogProgress(DeviceLayer, "Loading Java class references.");
+
+    chip::InitializeTracing();
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        // JniReferences::GetInstance().ThrowError(env, sFabricProviderCls, err);
+        JNI_OnUnload(jvm, reserved);
+    }
+
+    return err;
+}
+
+void AndroidChipFabricProviderJNI_OnUnload(JavaVM * jvm, void * reserved)
+{
+    ChipLogProgress(DeviceLayer, "ChipFabricProvider JNI_OnUnload() called");
+    chip::Platform::MemoryShutdown();
+}
+
+int GetFabricCount()
+{
+    return chip::Server::GetInstance().GetFabricTable().FabricCount();
+}
+
+jobject ReadFabricList(JNIEnv * env, jobject self)
+{
+    jobject jFabricList;
+    JniReferences::GetInstance().CreateArrayList(jFabricList);
+    jclass jFabricCls = env->FindClass("chip/appserver/Fabric");
+    VerifyOrExit(jFabricCls != nullptr, ChipLogError(NotSpecified, "unfind Class Fabric"));
+    {
+        for (auto & fabricInfo : Server::GetInstance().GetFabricTable())
+        {
+            // see src/app/clusters/operational-credentials-server/operational-credentials-server.cpp#ReadFabricsList
+            if (!fabricInfo.IsInitialized())
```

Comment:
Please see review comments on this in the other PR where this code was proposed.

Reply:
@bzbarsky-apple Hi, Boris, not found any comments about it in  https://github.com/project-chip/connectedhomeip/pull/12019, could you help to check this?

Reply:
Why are you looking in #12019?  See https://github.com/project-chip/connectedhomeip/pull/18546#discussion_r875958372

Reply:
Sorry, I misunderstand , @bzbarsky-apple I refer to src/app/clusters/operational-credentials-server/operational-credentials-server.cpp#ReadFabricsList function, it does the same thing, is it not necessary to do this?

Reply:
Yes, that code does not need this check either.

Reply:
I'll do a PR to fix the existing silliness.

Reply:
https://github.com/project-chip/connectedhomeip/pull/18772